pkg/util/release/virtual_strategy_test.go

```diff
@@ -0,0 +1,260 @@
+package release
+
+import (
+	"testing"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	testutil "github.com/bookingcom/shipper/pkg/util/testing"
+)
+
+func TestBuildVirtualStrategy(t *testing.T) {
+	var tests = []struct {
+		title                          string
+		maxSurge                       int
+		strategy                       *shipper.RolloutStrategy
+		expectedRolloutVirtualStrategy *shipper.RolloutVirtualStrategy
+	}{
+		{
+			"test building virtual strategy with maxSurge=30% (feature on)",
+			30,
+			&shipper.RolloutStrategy{
+				Steps: []shipper.RolloutStrategyStep{
+					{
+						Name:     "staging",
+						Capacity: shipper.RolloutStrategyStepValue{Incumbent: 100, Contender: 1},
+						Traffic:  shipper.RolloutStrategyStepValue{Incumbent: 100, Contender: 0},
+					},
+					{
+						Name:     "50/50",
+						Capacity: shipper.RolloutStrategyStepValue{Incumbent: 50, Contender: 50},
+						Traffic:  shipper.RolloutStrategyStepValue{Incumbent: 50, Contender: 50},
+					},
+					{
+						Name:     "full on",
+						Capacity: shipper.RolloutStrategyStepValue{Incumbent: 0, Contender: 100},
+						Traffic:  shipper.RolloutStrategyStepValue{Incumbent: 0, Contender: 100},
+					},
+				},
+			},
+			&shipper.RolloutVirtualStrategy{
+				Steps: []shipper.RolloutStrategyVirtualStep{
+					{
+						VirtualSteps: []shipper.RolloutStrategyStep{
+							{
+								Name: "0:  to staging",
```

Comment:
I wonder, why do we care about giving virtual strategy steps a name? This should be mostly an internal concern, and I don't expect it to aid debugging. Thoughts?

Reply:
I tend to agree, but when debugging I looked both at the numbers and at the title
after some more refactoring I will test it again and see if the title actually helps me